op-node/rollup/derive/channel_out_test.go

```diff
@@ -184,3 +219,94 @@ func TestBlockToBatchValidity(t *testing.T) {
 	_, _, err := BlockToSingularBatch(&rollupCfg, block)
 	require.ErrorContains(t, err, "has no transactions")
 }
+
+// TestSpanChannelOutCompressionOnlyOneBatch tests that the SpanChannelOut compression works as expected when there is only one batch
+// and it is larger than the target size. The single batch should be compressed, and the channel should now be full
+func TestSpanChannelOutCompressionOnlyOneBatch(t *testing.T) {
+	// target is larger than one batch, but smaller than two batches
+	target := uint64(300)
+	rng := rand.New(rand.NewSource(0x543331))
+	chainID := big.NewInt(rng.Int63n(1000))
+	txCount := 1
+	cout, err := NewSpanChannelOut(0, chainID, target)
+	require.NoError(t, err)
+
+	compressedLens := []int{}
+
+	// adding the first batch should not cause an error
+	singularBatch := RandomSingularBatch(rng, txCount, chainID)
+	_, err = cout.AddSingularBatch(singularBatch, 0)
+	compressedLens = append(compressedLens, cout.compressed.Len())
+	require.NoError(t, err)
+
+	// confirm compression was not skipped
+	require.Greater(t, compressedLens[0], 0)
+	// confirm the channel is full
+	require.ErrorIs(t, cout.FullErr(), ErrCompressorFull)
+}
+
+// TestSpanChannelOutCompressionUndo tests that the SpanChannelOut compression works as expected to not accept a second batch
+// if that batch would cause the channel to exceed the target size. The first batch should be compressed only
+// the channel should be compressed, full, and should not have the second batch
+func TestSpanChannelOutCompressionUndo(t *testing.T) {
+	// target is larger than one batch, but smaller than two batches
+	target := uint64(750)
+	rng := rand.New(rand.NewSource(0x543331))
+	chainID := big.NewInt(rng.Int63n(1000))
+	txCount := 1
+	cout, err := NewSpanChannelOut(0, chainID, target)
+	require.NoError(t, err)
+
+	compressedLens := []int{}
+	rlpLens := []int{}
+
+	// adding the first batch should not cause an error
+	singularBatch := RandomSingularBatch(rng, txCount, chainID)
+	_, err = cout.AddSingularBatch(singularBatch, 0)
+	compressedLens = append(compressedLens, cout.compressed.Len())
+	rlpLens = append(rlpLens, cout.activeRLP().Len())
+	require.NoError(t, err)
```

Comment:
You might as well do the assertion for the compressed length here instead of at the end. Then you don't need to collect the values in the compressedLens slice. I think that would actually be more readable.